catalog/rocks/validate.go

```diff
@@ -0,0 +1,97 @@
+package rocks
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/ident"
+)
+
+var (
+	validRefRegexp            = regexp.MustCompile(`^[^\s]+$`)
+	validBranchNameRegexp     = regexp.MustCompile(`^\w[-\w]*$`)
```

Comment:
Let's use either `BranchName` or `BranchID` everywhere.

Reply:
Also - isn't this a dup of `gravaler.reValidBranchID` & `graveler.reValidRepositoryID`?

Reply:
Moved them into this file